### PR TITLE
Reintroduce SAML SSO

### LINKED ISSUE #4880: reintroduce SAML SSO
it existed in 1.9 but got lost in transition to 2.0

### PR DESCRIPTION
**Pull request by devkiran**
## What does this PR do?

Fixes #4880

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by devkiran** [comment]
@deepakprabhakara @PeerRich

### Self-hosted
Added the "Single Sign-On" menu under the tab `Settings` > `Security.` The application hides this menu on the hosted version of the Cal. Since this is an enterprise feature, `CALCOM_LICENSE_KEY` is required to access this feature, and only the SAML admin (`SAML_ADMINS`) can access the page. A message instructing the user to set the values will be displayed if the `SAML_DATABASE_URL` or `SAML_ADMINS` is empty.

### Hosted
Added the "Single Sign-On" menu under each team. I have considered the following cases during the implementation into the team. Please let me know if I have covered every case. This menu is only visible on the hosted version and will redirect to `/404` if the user tries accessing it on the self-hosted version. At the moment, only a user with a `PRO` plan can access the page, and the member should either `ADMIN` or `OWNER` of the team.

File: packages/features/ee/sso/lib/saml.ts [hunk 1]
```diff
@@ -57,3 +58,49 @@ export const samlTenantProduct = async (prisma: PrismaClient, email: string) =>
     product: samlProductID,
   };
 };
+
+export const canAccess = async (
```
Comment (devkiran) [inline packages/features/ee/sso/lib/saml.ts:62]: This method has been altered to send back appropriate messages in each case. This message will be displayed on the UI when TRPC throws an error https://github.com/calcom/cal.com/pull/4938/files#diff-015067416d87bbb0a3405527c31019007de06a40b377883a98b6e32623c2d0e5R25

**Comment by PeerRich** [comment]
hey @devkiran, thank you for the update. would you mind fixing the last merge conflict and we'll then review & merge 🙏

**Review by zomars — APPROVED** [review]
Simplified api endpoints and made some file organization fixes. Code LGTM testing on staging.

**Comment by devkiran** [comment]
@zomars thanks for helping me out with the merge and conflict.

**Pull request merged**